packages/flutter_migrate/lib/src/compute.dart

```diff
@@ -0,0 +1,1014 @@
+}
+
+String platformToSubdirectoryPrefix(SupportedPlatform platform) {
+  switch (platform) {
+    case SupportedPlatform.android:
+      return 'android';
+    case SupportedPlatform.ios:
+      return 'ios';
+    case SupportedPlatform.linux:
+      return 'linux';
+    case SupportedPlatform.macos:
+      return 'macos';
+    case SupportedPlatform.web:
+      return 'web';
+    case SupportedPlatform.windows:
+      return 'windows';
+    case SupportedPlatform.fuchsia:
+      return 'fuchsia';
+    case SupportedPlatform.root:
+      return 'root';
+  }
+}
+
+/// Computes the changes that migrates the current flutter project to the target revision.
+///
+/// This is the entry point to the core migration computations.
+///
+/// This method attempts to find a base revision, which is the revision of the Flutter SDK
+/// the app was generated with or the last revision the app was migrated to. The base revision
+/// typically comes from the .metadata, but for legacy apps, the config may not exist. In
+/// this case, we fallback to using the revision in .metadata, and if that does not exist, we
+/// use the target revision as the base revision. In the final fallback case, the migration should
+/// still work, but will likely generate slightly less accurate merges.
+///
+/// Operations the computation performs:
+///
+///  - Parse .metadata file
+///  - Collect revisions to use for each platform
+///  - Download each flutter revision and call `flutter create` for each.
+///  - Call `flutter create` with target revision (target is typically current flutter version)
+///  - Diff base revision generated app with target revision generated app
+///  - Compute all newly added files between base and target revisions
+///  - Compute merge of all files that are modified by user and flutter
+///  - Track temp dirs to be deleted
+///
+/// Structure: This method builds upon a MigrateResult instance
+Future<MigrateResult?> computeMigration({
+  bool verbose = false,
+  FlutterProject? flutterProject,
+  String? baseAppPath,
+  String? targetAppPath,
+  String? baseRevision,
+  String? targetRevision,
+  bool deleteTempDirectories = true,
+  List<SupportedPlatform>? platforms,
+  bool preferTwoWayMerge = false,
+  bool allowFallbackBaseRevision = false,
+  required FileSystem fileSystem,
+  required Logger logger,
+  required MigrateUtils migrateUtils,
+  required FlutterToolsEnvironment environment,
+}) async {
+  flutterProject ??= FlutterProject.current(fileSystem);
+
+  logger.printStatus(
+      'Computing migration - this command may take a while to complete.');
+  // We keep a spinner going and print periodic progress messages
+  // to assure the developer that the command is still working due to
+  // the long expected runtime.
+  final Status status = logger.startSpinner();
+  status.pause();
+  logger.printStatus('Obtaining revisions.',
+      indent: 2, color: TerminalColor.grey);
+  status.resume();
```

Comment:
Having UI and core logic intermixed is going to make this hard to maintain over time (and to extend to other uses, like a GUI tool). I would strongly recommend having a state system that this communicates to, and have that state drive UI from completely separate code.